products/error_tracking/frontend/components/ExceptionAttributesPreview/ReleasesPreview/releasePreviewLogic.ts

```diff
@@ -0,0 +1,69 @@
+import { kea, path } from 'kea'
+import { loaders } from 'kea-loaders/lib'
+
+import api from 'lib/api'
+import 'lib/components/Errors/stackFrameLogic'
+import { ErrorTrackingStackFrame, RawEventExceptionRelease } from 'lib/components/Errors/types'
+
+import type { releasePreviewLogicType } from './releasePreviewLogicType'
+
+export const releasePreviewLogic = kea<releasePreviewLogicType>([
+    path([
+        'products',
+        'error_tracking',
+        'frontend',
+        'components',
+        'ExceptionAttributesPreview',
+        'ReleasesPreview',
+        'releasePreviewLogic',
+    ]),
+
+    loaders(() => ({
+        release: [
+            undefined as RawEventExceptionRelease | undefined,
+            {
+                loadRelease: async (dto: {
+                    // we have exceptionReleases from the already loaded event. Cymbal enriches event with that data
+                    exceptionReleases?: RawEventExceptionRelease[]
+                    frames?: ErrorTrackingStackFrame[]
+                }) => {
+                    // we can't do anything if there are neither frames nor existing releases
+                    if (!dto.frames && !dto.exceptionReleases) {
+                        return undefined
+                    }
+
+                    // if there is only one associated release, there is no need to look for the kaboom frame
+                    // because we would find the same release anyways. This is a big performance win. I calculated
+                    // how many events are related to how many releases and 95% events in the last 30 days are related to
+                    // at most 1 release
```

Comment:
We can shorten this comment (and probably remove most others in the file) I think the code is relatively explanatory